### DISCUSSION TITLE
Grouping related data in a table in QuestPdf

### DISCUSSION THREAD

**Discussion by franks1** [QuestPDF/QuestPDF#523] [General]
Can i group related data with QuestPdf?
A typical example will be to create a table for a group of data having the group as a header in the content and generating another table of related data in a separate group using a custom data source.

**Comment by girlpunk** [comment]
There's no direct data manipulation in QuestPDF, but you can achieve similar results by doing the grouping yourself (probably using [LINQ](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.groupby?view=net-8.0)) and then inserting headers into a table as you iterate through groups and data within them.
